### ISSUE TITLE
The model cannot be fixed by the seed

### ISSUE THREAD

**Issue by tingtingring** [MAGICS-LAB/DNABERT_2#139]
We use DNABERT2 to perform downstream tasks. However, after several trials, we find that the model cannot be fixed by the seed. That is to say, every time we run the model, the results vary even when the same seed is used. We checked the whole model and did not find any problems that could affect the randomness.So I am here to ask this question.Hope you can help me to deal with this.

**Comment by Zhihan1996** [comment]
Hey, there could be varies sources of randomness that affect the final model performance. If you want absolute identical results in each training, please refer to the official pytorch tutorial. https://pytorch.org/docs/stable/notes/randomness.html.

**Comment by tingtingring** [comment]
We found that if we set the deterministic parameter in set_seed from transformers to True, the model will throw an error. I was wondering where your model uses this.It is worth mentioning that, apart from transformers, the seeds for both torch and numpy have already been fixed.
